holoviews/core/spaces.py

```diff
@@ -788,6 +804,105 @@ def dynamic_redim(obj):
         return Dynamic(redimmed, shared_data=True, operation=dynamic_redim)
 
 
+    def collate(self):
+        """
+        Collation allows collapsing DynamicMaps with invalid nesting
+        hierarchies. This is particularly useful when defining
+        DynamicMaps returning an (Nd)Layout or GridSpace
+        type. Collating will split the DynamicMap into of individual
+        DynamicMaps. Note that the composite object has to be of
+        consistent length and types for this to work
+        correctly. Associating streams with specific viewables in the
+        returned container declare a stream_mapping on the DynamicMap
+        Callable during instantiation.
+        """
+        # Initialize
+        if self.last is not None:
+            pass
+        else:
+            self[self._initial_key()]
```

Comment:
I'm not saying this option would be used a lot (in fact I'm counting on the opposite). I just think we can offer the option to disable some slightly surprising behavior. Won't calling ``self[self._initial_key()]`` change the cache? It must do as you are then using ``last``.

Maybe you should clone self and then use ``self[self._initial_key()]`` which would then not result in a side-effect. Then, if I understand correctly, the use of ``.last`` is just to get an example of the sort of thing the callback is returning...